pkg/proxy/backend/backend_conn_mgr.go

```diff
@@ -636,13 +636,13 @@ func (mgr *BackendConnManager) Close() error {
 	var connErr error
 	var addr string
 	mgr.processLock.Lock()
-	if backendIO := mgr.backendIO.Swap(nil); backendIO != nil {
+	if backendIO := mgr.backendIO.Load(); backendIO != nil {
 		addr = backendIO.RemoteAddr().String()
 		connErr = backendIO.Close()
 	}
-	mgr.processLock.Unlock()
-
 	handErr := mgr.handshakeHandler.OnConnClose(mgr)
```

Comment:
Maybe we can just invoke `OnConnClose` before L639?

Reply:
I think OnHandshake is only called after the task is completed, and it also has processLock protection. just move before L639?

Reply:
done